### ISSUE TITLE
library path is missing in arpack-ng-config.cmake

### ISSUE THREAD

**Issue by RyanBernX** [opencollab/arpack-ng#297]
## Expected behavior

arpack-ng-config.cmake works properly.

## Actual behavior

cmake cannot find `-larpack` when linking

## Where/how to reproduce the problem

- arpack-ng: e3bdd3b8db727a10551c06959e45e866a6b8d8fc
- OS: ubuntu 20.04
- compiler: gcc
- cmake 3.15

## Steps to reproduce the problem

- Install arpack-ng to a non-default location (e.g. `/opt/ARPACK`)
- In another cmake project, use arpack-ng as suggested in `README.md`
- Invoke cmake with `cmake -DCMAKE_PREFIX_PATH=/opt/ARPACK ..` 

## Error message

```
/usr/bin/ld: cannot find -larpack
collect2: error: ld returned 1 exit status
```

## Notes, remarks
It seems that the library path is missing in `arpack-ng-config.cmake`, causing gcc only to find `libarpack.so` in default locations.

More comments about using cmake with arpack-ng:
1) It may be better to set C99 standard in `CMakeLists.txt`, as some old compilers (e.g. gcc-4.8.5) do not enable this by default.
2) In `README.md`, `target_include_directories(main PUBLIC ARPACK::ARPACK)` seems wrong. It makes no sense to use an interface target as an include directory. Consequently, cmake will generate something like `-I/path/to/cmakelists/ARPACK::ARPACK` which is definitely not we want. In this case, a single `target_link_libraries(main ARPACK::ARPACK)` may suffice.

**Comment by fghoussen** [comment]
Feel free to propose a PR. CI is broken but if PR is tested OK at your side and only involves `*.cmake` it may be merged.
The best solution for me would still be https://github.com/opencollab/arpack-ng/pull/259#issuecomment-603187398 but as @turboencabulator says 'no idea how to do this' https://github.com/opencollab/arpack-ng/pull/259#issuecomment-603581189

**Comment by RyanBernX** [comment]
@DavidAce has committed a newer version of `.cmake.in` 64cc908 . Could you please take a look at https://github.com/opencollab/arpack-ng/pull/296#issuecomment-797506948 ?

**Comment by fghoussen** [comment]
should be fixed by #350
